src/search.js

```diff
@@ -65,10 +72,17 @@ angular
 				 * Updates the model that the user passed in (attributes#param)
 				 */
 				update: function() {
-					scope.search({
-						$param: scope.param
-					});
+					// Do not allow search untill search character meet the minimum length requirement
+					if(scope.minSearchLength && scope.param !== '' && scope.param.length < scope.minSearchLength){
+						scope.searchValidation = true;
+					} else {
+						scope.searchValidation = false;
+						scope.search({
+							$param: scope.param
+						});
+					}
 					scope.dirty = false;
+					scope.$apply();
```

Comment:
apply is required because we are attaching search input change event through `.on` jquery method not using angular event so it want update the scope variable `searchValidation` in view